rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -0,0 +1,202 @@
+        At the moment it does not decode the fee in ETH. For example:
+        etherscan.io/tx/0xeeea20b39f157fe59fa4904fd4b62f8971188b53d05c6831e0ed67ee157e40c2"""
+        if not self.base.is_tracked(sender) and not self.base.is_tracked(receiver):
+            return DEFAULT_DECODING_OUTPUT
+
+        out_event: EvmEvent | None = None
+        in_event: EvmEvent | None = None
+        partial_refund_event: EvmEvent | None = None
+        fee_event: EvmEvent | None = None
+        for event in context.decoded_events:
+            if sender != event.location_label:  # in this case, it's not a valid send/receive event
+                continue
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND or
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):  # find the send event
+                event.counterparty = CPT_PARASWAP
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.address = self.router_address
+                out_event = event
+            elif (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.RECEIVE or
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):  # find the receive event
+                event.counterparty = CPT_PARASWAP
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.notes = f'Receive {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} as the result of a swap in paraswap'  # noqa: E501
+                event.address = self.router_address
+                if in_event is None:
+                    in_event = event
+                else:  # if another in_event is found, then it is the return of in_asset
+                    partial_refund_event = event
+
+        if in_event is None or out_event is None:
+            log.error(f'Could not find the corresponding events when decoding paraswap swap {context.transaction.tx_hash.hex()}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        if partial_refund_event is not None:  # if some in_asset is returned back
+            out_event.balance.amount -= partial_refund_event.balance.amount  # adjust the amount
+            context.decoded_events.remove(partial_refund_event)  # and remove it from the list
+        out_event.notes = f'Swap {out_event.balance.amount} {out_event.asset.resolve_to_asset_with_symbol().symbol} in paraswap'  # noqa: E501
+
+        fee_raw = fee_asset = None  # extract the fee info
+        for context_log in context.all_logs:
```

Comment:
This is just a normal log_event

```suggestion
        for log_event in context.all_logs:
```